plugins/inputs/chrony/chrony.go

```diff
@@ -38,6 +41,27 @@ func (*Chrony) SampleConfig() string {
 	return sampleConfig
 }
 
+// dialUnix opens an unixgram connection with chrony
+func dialUnix(address string) (*net.UnixConn, error) {
+	dir := path.Dir(address)
+	local := path.Join(dir, fmt.Sprintf("chrony-telegraf-%s.sock", uuid.New().String()))
+	// XXX TODO: remove the file at the end
+	conn, err := net.DialUnix("unixgram",
+		&net.UnixAddr{Name: local, Net: "unixgram"},
+		&net.UnixAddr{Name: address, Net: "unixgram"},
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if err := os.Chmod(local, 0666); err != nil {
+		return nil, err
+	}
```

Comment:
@jdstrand can you please review this and the conversation in the PR please? Thanks

Reply:
@powersj - I can confirm that locally `/run/chrony` is `_chrony:_chrony` 0750 so having the socket set to `0666` would not be a problem on this system. I'm puzzled why `0666` makes things work when the directory presumably would also have these restrictive permissions so the socket should've been denied regardless of the socket permissions. Where 0666 works and 0660 or 0600 doesn't, can someone confirm the ownership and permissions on the `/run/chrony` directory, what user telegraf is running as and what group membership that user has?

I'm inclined for our code to create the socket with the minimal privileges that telegraf needs and not make assumptions about the run directory, but until we understand why 0666 was needed, I can't make a recommendation.

(AFAIK, telegraf doesn't have the concept of dropping privileges (please correct me) and it is simply started as the desired user. So, for `root` I wouldn't expect `0666` to have been needed and for `telegraf` I would've expected the socket creation to have failed since only the `_chrony` *owner* has write to the directory to create the socket).

Reply:
@Frankkkkk could you share the following on your system:

1. the ownership and permissions on the /run/chrony
2. what user telegraf was running as
3. the groups that user belongs to?

@jdstrand in terms of ownership this is what I see:

On Ubuntu Jammy:

```shell
root@j1:~# stat /run/chrony
  File: /run/chrony
  Size: 80        	Blocks: 0          IO Block: 4096   directory
Device: 61h/97d	Inode: 334         Links: 2
Access: (0750/drwxr-x---)  Uid: (  106/ _chrony)   Gid: (  113/ _chrony)
Access: 2024-07-08 20:09:21.502577346 +0000
Modify: 2024-07-08 20:09:21.502577346 +0000
Change: 2024-07-08 20:09:21.502577346 +0000
 Birth: 2024-07-08 20:09:21.502577346 +0000
```

On Arch:

```shell
❯ stat /run/chrony
  File: /run/chrony
  Size: 80        	Blocks: 0          IO Block: 4096   directory
Device: 0,23	Inode: 5275        Links: 2
Access: (0750/drwxr-x---)  Uid: (  956/  chrony)   Gid: (  956/  chrony)
Access: 2024-07-08 14:10:06.419406121 -0600
Modify: 2024-07-08 14:10:06.422520696 -0600
Change: 2024-07-08 14:10:06.422520696 -0600
 Birth: 2024-07-08 14:10:06.419406121 -0600
```